src/util.cc

```diff
@@ -475,4 +480,19 @@ void SetConstructorFunction(Local<v8::Context> context,
   that->Set(context, name, tmpl->GetFunction(context).ToLocalChecked()).Check();
 }
 
+const char* FindSingleExecutableCode(size_t* size) {
```

Comment:
Having this being called multiple times in several places feels like something unoptimised. Wouldn't it make sense to transform it into a memoized call? Such that its execution only happens once, and the cached value is returned?

I also feel we should only call this IF the `--compile` arg was provided. For example, on your argv check you're checking against finding the single executable code, instead of checking the args 🤔

Reply:
> memoized call

This is already caching the values in the `single_executable_application_code_loaded`, `single_executable_application_size` and `single_executable_application_code` variables declared at the top, right? It forwards the call to `postject_find_resource()` only if `single_executable_application_code_loaded` is `false`, otherwise it returns the values of `single_executable_application_size` and `single_executable_application_code`.

> we should only call this IF the --compile arg was provided

This function is used to check if a resource has been injected into node to decide if the executable is supposed to be treated as a single executable, in which case it would use the returned buffer. Otherwise, it would operate as good old node. So we would need to run this function even if `--compile` is not passed to check if the current executable is an SEA or just plain Node.js.

Reply:
> This function is used to check if a resource has been injected into node to decide if the executable is supposed to be treated as a single executable, in which case it would use the returned buffer. Otherwise, it would operate as good old node. So we would need to run this function even if --compile is not passed to check if the current executable is an SEA or just plain Node.js.

Interesting. So you're saying we should still treat is a SEA if I have `node << bunch of something to be injected` even if we're not passing the flag?